protocol/wasmbinding/msg_plugin.go

```diff
@@ -3,136 +3,112 @@ package wasmbinding
-	return func(old wasmkeeper.Messenger) wasmkeeper.Messenger {
-		return &CustomMessenger{
-			wrapped: old,
-			sending: sending,
-			clob:    clob,
-		}
-	}
-}
-
-type CustomMessenger struct {
-	wrapped wasmkeeper.Messenger
-	sending *sendingkeeper.Keeper
-	clob    *clobkeeper.Keeper
-}
-
-var _ wasmkeeper.Messenger = (*CustomMessenger)(nil)
-
 // DispatchMsg executes on the contractMsg.
-func (m *CustomMessenger) DispatchMsg(ctx sdk.Context, contractAddr sdk.AccAddress, contractIBCPortID string, msg wasmvmtypes.CosmosMsg) ([]sdk.Event, [][]byte, error) {
-	if msg.Custom != nil {
-		// only handle the happy path where this is really creating / minting / swapping ...
-		// leave everything else for the wrapped version
-		var contractMsg bindings.SendingMsg
-		// print the custom message in string format
-		if err := json.Unmarshal(msg.Custom, &contractMsg); err != nil {
-			return nil, nil, errorsmod.Wrap(err, "Error Unmarshalling Custom Message")
-		}
-		if contractMsg.CreateTransfer != nil {
-			return m.createTransfer(ctx, contractAddr, contractMsg.CreateTransfer)
-		}
-		if contractMsg.DepositToSubaccount != nil {
-			return m.depositToSubaccount(ctx, contractAddr, contractMsg.DepositToSubaccount)
-		}
-		if contractMsg.WithdrawFromSubaccount != nil {
-			return m.withdrawFromSubaccount(ctx, contractAddr, contractMsg.WithdrawFromSubaccount)
-		}
-		if contractMsg.PlaceOrder != nil {
-			return m.placeOrder(ctx, contractAddr, contractMsg.PlaceOrder)
-		}
-		if contractMsg.CancelOrder != nil {
-			return m.cancelOrder(ctx, contractAddr, contractMsg.CancelOrder)
-		}
-		return nil, nil, wasmvmtypes.InvalidRequest{Err: "Unknown custom message"}
+func CustomEncoder(sender sdk.AccAddress, msg json.RawMessage) ([]sdk.Msg, error) {
+	var customMessage bindings.DydxCustomWasmMessage
+	if err := json.Unmarshal(msg, &customMessage); err != nil {
+		return []sdk.Msg{}, wasmvmtypes.InvalidRequest{Err: "Error parsing DydxCustomWasmMessage"}
 	}
-	return m.wrapped.DispatchMsg(ctx, contractAddr, contractIBCPortID, msg)
-}
-
-func (m *CustomMessenger) createTransfer(ctx sdk.Context, contractAddr sdk.AccAddress, createTransfer *sendingtypes.MsgCreateTransfer) ([]sdk.Event, [][]byte, error) {
-	if createTransfer == nil {
-		return nil, nil, wasmvmtypes.InvalidRequest{Err: "Invalid create transfer request: No transfer data provided"}
+	switch {
+	case customMessage.DepositToSubaccount != nil:
+		return EncodeDepositToSubaccount(sender, customMessage.DepositToSubaccount)
+	case customMessage.WithdrawFromSubaccount != nil:
+		return EncodeWithdrawFromSubaccount(sender, customMessage.WithdrawFromSubaccount)
+	case customMessage.PlaceOrder != nil:
+		return EncodePlaceOrder(sender, customMessage.PlaceOrder)
+	case customMessage.CancelOrder != nil:
+		return EncodeCancelOrder(sender, customMessage.CancelOrder)
+	default:
+		return nil, wasmvmtypes.InvalidRequest{Err: "Unknown Dydx Wasm Message"}
 	}
```

Comment:
Can we add a TODO item in Linear to add E2E test for verifying that contract cannot send `DepositToSubaccount` message on behalf of another address? Up to you on what's the best way to test this